.ci/checker-framework.groovy

```diff
@@ -113,27 +113,43 @@ private static int checkCheckerFrameworkReport(final String profile) {
 private static List<List<String>> getCheckerFrameworkErrors(final String profile) {
     final List<String> checkerFrameworkLines = new ArrayList<>()
     final String command = "mvn -e --no-transfer-progress clean compile -P${profile}"
-    final Process process = getOsSpecificCmd(command).execute()
-    process.in.eachLine { final line ->
-        checkerFrameworkLines.add(line)
-        println(line)
+    final ProcessBuilder processBuilder = new ProcessBuilder(getOsSpecificCmd(command).split(' '))
+    processBuilder.redirectErrorStream(true)
+    final Process process = processBuilder.start()
+
+    final BufferedReader reader = new BufferedReader(new InputStreamReader(process.inputStream))
+    String lineFromReader
+    while ((lineFromReader = reader.readLine()) != null) {
+        println(lineFromReader)
+        checkerFrameworkLines.add(lineFromReader)
     }
     process.waitFor()
+
+    for (int index = checkerFrameworkLines.size() - 1; index >= 0; --index) {
+        final String curLine = checkerFrameworkLines.get(index)
+        if (curLine.startsWith('[ERROR]') || curLine.startsWith('[INFO]')) {
+            checkerFrameworkLines.remove(index)
```

Comment:
We removing elements from same collection as we iterate, and recount index - it is a problem, you can miss items.
Better to have new collection for result.

Reply:
Yes, @Vyom-Yadav why add lines to collection just to be removed anyway and iterate through collection twice? Above, we are already iterating over lines, why not add only `[WARNING]` lines to collection after being trimmed?

Reply:
> We removing elements from same collection as we iterate, and recount index - it is a problem, you can miss items.

That's why I am iterating in reverse order, already examined items won't be affected and the upcoming items index won't change.

Reply:
> Yes, @Vyom-Yadav why add lines to collection just to be removed anyway and iterate through collection twice? Above, we are already iterating over lines, why not add only `[WARNING]` lines to collection after being trimmed?

I initially did that, but I was facing https://github.com/checkstyle/checkstyle/pull/12366#issuecomment-1303009194, so I tired this one to see if it makes any difference, reverting back to filtering in a single iteration appraoch.

Reply:
Done.